rspotify-model/src/idtypes.rs

```diff
@@ -156,26 +154,26 @@ pub trait Id: Send + Sync {
     /// Examples: `spotify:album:6IcGNaXFRf5Y1jc7QsE9O2`,
     /// `spotify:track:4y4VO05kYgUTo2bzbox1an`.
     fn uri(&self) -> String {
-        format!("spotify:{}:{}", self._type(), self.id())
+        format!("spotify:{}:{}", Self::TYPE, self.id())
     }
 
     /// Full Spotify object URL, can be opened in a browser
     ///
     /// Examples: `https://open.spotify.com/track/4y4VO05kYgUTo2bzbox1an`,
     /// `https://open.spotify.com/artist/2QI8e2Vwgg9KXOz2zjcrkI`.
     fn url(&self) -> String {
-        format!("https://open.spotify.com/{}/{}", self._type(), self.id())
+        format!("https://open.spotify.com/{}/{}", Self::TYPE, self.id())
     }
 
-    /// Parse Spotify id from string slice
+    /// Parse Spotify Id from string slice.
     ///
     /// A valid Spotify object id must be a non-empty string with valid
     /// characters.
     ///
     /// # Errors:
     ///
     /// - `IdError::InvalidId` - if `id` contains invalid characters.
-    fn from_id(id: &str) -> Result<Self, IdError>
+    fn from_id(id: &'a str) -> Result<Self, IdError>
```

Comment:
This function should probably make use of the new `id_is_valid`, otherwise the tests for user ids fail.